docs/developers/architecture/app_model.md

```diff
@@ -265,11 +423,162 @@ be used easily as a command in a menu, or bound to a keybinding.  It is up to
+layers"). They are defined in
+[`napari/_app_model/context`](https://github.com/napari/napari/tree/main/napari/_app_model/context).
+An instance of each class is created and saved as an attribute of
+{class}`~napari.components.LayerList`. The
+{meth}`~napari._app_model.context._context_keys.ContextNamespace.update`
+methods of each are then connected to events that change the
+{class}`~napari.components.LayerList` or selected {class}`~napari.components.LayerList`.
+
+Variables in these classes can be used in expressions
+in the [`Action`](app-model-actions) `enablement` field and
+menus {class}`~app_model.types.MenuRule`'s `when` field (see [](app-model-menus
+for details)).
+
+Currently, the {class}`app_model.Application` class does not have a
+context registry but it may in future (more details can be found
+[here](https://github.com/pyapp-kit/app-model/pull/142#issuecomment-1792405334)).
+napari therefore needs to manually update the context via
+{meth}`~app_model.backends.qt.QMenuItemAction.update_from_context`. For menu bar items,
+we connect {meth}`~app_model.backends.qt.QMenuItemAction.update_from_context` to the
+`aboutToShow` event of each menu bar `QModelMenu` instance,
+in {meth}`napari.window.Window._add_menus`.
+
+# Migration from action manager
+
 ## Motivation & Future Vision
 
-While it's certainly possible that there will be cases where this abstraction proves to be a bit more annoying than the previous "procedural" approach, there are a number of motivations for adopting this abstraction.
+While it's certainly possible that there will be cases where this abstraction
+in app-model proves to be a bit more annoying than the previous "procedural" approach,
+there are a number of motivations for adopting this abstraction.
 
 1. It gives us an abstraction layer on top of Qt that will make it much easier to explore different application backends (such as a web-based app, etc..)
 1. It's easier to test: `app-model` can take care of making sure that commands, menus, keybindings, and actions are rendered, updated, and triggered correctly, and napari can focus on testing the napari-specific logic.
 1. It's becomes **much** easier to add & remove contributions from plugins if our internal representation of a command, menu, keybinding is similar to the schema that plugins use. The previous procedural approach made this marriage much more cumbersome.
 1. **The Dream**: The unification of napari commands and plugin commands into a registry that can execute commands in response to user input provides an excellent base for "recording" a user workflow.  If all GUI user interactions go through dependency-injected commands, then it becomes much easier to export a script that reproduces a set of interactions.
+
+(app-model-action-manager-differences)=
+
+## app-model vs action manager implementation differences
+
+App-model and action manager differ in when actions are executed.
+In app-model, actions are defined executed when they
```

Comment:
> actions are defined executed when

This is confusing to me, either one of those should be dropped (executed?) or there is an "and" missing?

Reply:
Oh yeah I mucked that up. I think I mean 'executed'? (because define just means which file they are written in right??). When a function is connected to an event or registered to the `app`, the function itself is 'run' right? I mean we import the e.g., view menu actions and that import runs the functions inside the file where the view menu actions are define?!

What is the best phrasing here?

cc @DragaDoncila

Reply:
I would not use `executed` in this context. I would say "defined" when referring to the file they are literally written in, and `registered` when they are added to the app-model/action_manager registry.

Reply:
Yeah good point. I've remembered what I was thinking about when I wrote this -> plugins menu actions used to be registered later when we create the plugins menu (when we know we have Qt). Now we register plugin actions much earlier, when we don't know if we have Qt. (Thus we've had to do 'safe register')

Reply:
Done @DragaDoncila, let me know what you think, thanks!